odmlui/Editor.py

```diff
@@ -484,6 +536,78 @@ def map_tab(self, tab):
+
+    @gui_action("odMLTablesConvert", tooltip="Convert document to xls or csv",
+                label="odMLTablesConverter", stock_id="INM6-convert-odml",
+                accelerator="<control>C")
+    def on_convert(self, action):
+        if not self.current_tab.file_uri or self.current_tab.is_modified:
+            self._info_bar.show_info("Please validate and save "
+                                     "your document before starting odMLTables.")
+        elif self.odml_tables_available:
+            # odmltables accepts only odml xml files with an '.odml' file ending.
+            tail = os.path.split(uri_to_path(self.current_tab.file_uri))[1]
+            tmp_file = os.path.join(CACHE_DIR, ("%s.odml" % tail))
+            odml.fileio.save(self.current_tab.document, tmp_file)
+            os.system("odmltables -w convert -f %s &" % tmp_file)
+        else:
+            self._info_bar.show_info("You need odMLTables (v%s or newer) "
+                                     "installed to run this feature." %
+                                     ODMLTABLES_VERSION)
+
+    @gui_action("odMLTablesFilter", tooltip="Filter document contents",
+                label="odMLTablesFilter", stock_id="INM6-filter-odml",
+                accelerator="<control>F")
+    def on_filter(self, action):
+        if not self.current_tab.file_uri or self.current_tab.is_modified:
+            self._info_bar.show_info("Please validate and save "
+                                     "your document before starting odMLTables.")
+        elif self.odml_tables_available:
+            # odmltables accepts only odml xml files with an '.odml' file ending.
+            tail = os.path.split(uri_to_path(self.current_tab.file_uri))[1]
+            tmp_file = os.path.join(CACHE_DIR, ("%s.odml" % tail))
+            odml.fileio.save(self.current_tab.document, tmp_file)
+            os.system("odmltables -w filter -f %s &" % tmp_file)
+        else:
+            self._info_bar.show_info("You need odMLTables (v%s or newer) "
+                                     "installed to run this feature." %
+                                     ODMLTABLES_VERSION)
+
+    @gui_action("odMLTablesMerge", tooltip="Merge odML documents",
+                label="odMLTablesMerge", stock_id="INM6-merge-odml",
+                accelerator="<control>M")
+    def on_merge(self, action):
+        if not self.current_tab.file_uri or self.current_tab.is_modified:
+            self._info_bar.show_info("Please validate and save "
+                                     "your document before starting odMLTables.")
+        elif self.odml_tables_available:
+            # odmltables accepts only odml xml files with an '.odml' file ending.
+            tail = os.path.split(uri_to_path(self.current_tab.file_uri))[1]
+            tmp_file = os.path.join(CACHE_DIR, ("%s.odml" % tail))
+            odml.fileio.save(self.current_tab.document, tmp_file)
+            os.system("odmltables -w merge &")
```

Comment:
Does the merge function not take file arguments?

Reply:
Yes, you can provide one or two files name using -f (--file).

Reply:
Then we shall provide the currently open document as the first file!